### PR TITLE
Detect whether the Debug Extension was built for dev or release

### PR DESCRIPTION
**Pull request by elliette**
Detects whether the Debug Extension was built for dev or release. 

In dev mode:
- only logs to the console in dev mode (unless specified otherwise)
- uses the dev icon instead of the prod icon 

This PR also moves the `console.log` implementation into `logger` and makes it private, so that the only way to log to the console is with the `debugLog` utilities.

### CODE REVIEW

File: dwds/debug_extension_mv3/tool/update_dev_files.dart [hunk 1]
```diff
@@ -19,7 +19,11 @@ Future<void> _updateManifestJson() async {
   return _transformDevFile(manifestJson, (line) {
     if (_matchesKey(line: line, key: 'name')) {
       return [
-        line,
+        _newKeyValue(
+          oldLine: line,
+          newKey: 'name',
+          newValue: '[DEV] MV3 Dart Debug Extension',
```
Comment (annagrin) [inline dwds/debug_extension_mv3/tool/update_dev_files.dart:25]: Thanks! That is really helpful to distinguish them from each other!

File: dwds/debug_extension_mv3/web/logger.dart [hunk 2]
```diff
@@ -7,41 +7,71 @@ library logger;
...
   error,
 }
 
-debugLog(String msg, {String? prefix}) {
+debugLog(
+  String msg, {
+  String? prefix,
+  bool devOnly = true,
+}) {
   _log(msg, prefix: prefix);
```
Comment (annagrin) [inline dwds/debug_extension_mv3/web/logger.dart:23]: needs to pass `devOnly` to `_log`?

File: dwds/debug_extension_mv3/web/logger.dart [hunk 3]
```diff
@@ -7,41 +7,71 @@ library logger;
...
 }
 
-debugWarn(String msg, {String? prefix}) {
+debugWarn(
+  String msg, {
+  String? prefix,
+  bool devOnly = true,
+}) {
   _log(msg, prefix: prefix, level: _LogLevel.warn);
```
Comment (annagrin) [inline dwds/debug_extension_mv3/web/logger.dart:31]: same as above

File: dwds/debug_extension_mv3/web/logger.dart [hunk 4]
```diff
@@ -7,41 +7,71 @@ library logger;
...
 }
 
-debugError(String msg, {String? prefix}) {
+debugError(
+  String msg, {
+  String? prefix,
+  bool devOnly = true,
```
Comment (annagrin) [inline dwds/debug_extension_mv3/web/logger.dart:37]: same as above

File: dwds/debug_extension_mv3/web/logger.dart [hunk 5]
```diff
@@ -7,41 +7,71 @@ library logger;
...
 
-void _log(String msg, {_LogLevel? level, String? prefix}) {
-  if (!_enableDebugLogging) return;
+void _log(
+  String msg, {
+  bool devOnly = true,
+  _LogLevel? level,
+  String? prefix,
+}) {
+  if (devOnly && !isDevMode()) return;
```
Comment (annagrin) [inline dwds/debug_extension_mv3/web/logger.dart:48]: Would it be reasonable to rename `devOnly` to `verbose` and maybe flip the default to false?

**Review by annagrin — APPROVED** [review]
LGTM with comments.

File: dwds/debug_extension_mv3/web/logger.dart (hunk 2 above)
Comment (elliette) [reply dwds/debug_extension_mv3/web/logger.dart]: Ah good catch!

File: dwds/debug_extension_mv3/web/logger.dart (hunk 5 above)
Comment (elliette) [reply dwds/debug_extension_mv3/web/logger.dart]: Sg, switched to verbose!

**Review by annagrin — APPROVED** [review]
LGTM!

**Pull request merged**